azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/configuration/events/CallCompositeEventsHandler.kt

```diff
@@ -8,19 +8,23 @@ import com.azure.android.communication.ui.calling.models.CallCompositeErrorEvent
 import com.azure.android.communication.ui.calling.models.CallCompositeRemoteParticipantJoinedEvent
 
 internal class CallCompositeEventsHandler {
-    private var errorHandlers: CallCompositeEventHandler<CallCompositeErrorEvent>? = null
-    private var remoteParticipantJoinedHandler: CallCompositeEventHandler<CallCompositeRemoteParticipantJoinedEvent>? =
-        null
+    private val errorHandlers = mutableSetOf<CallCompositeEventHandler<CallCompositeErrorEvent>>()
+    private val remoteParticipantJoinedHandler =
```

Comment:
For future, as there are more of these handler types, it may be worth it to generalize this a tiny bit into a "handler registry" class of sorts? Something defined against `handlers: mutableSetOf<T>`, with `add`/`remove`/`notify` methods and proper synchronization.

Reply:
Thank you, we will consider it in the future. We may also use azure core future implementation for events.